wheatley/bot.py

```diff
@@ -171,24 +179,43 @@ def look_to_has_been_called(self, call_time: float) -> None:
         self.row_generator = self.next_row_generator or self.row_generator
         self.next_row_generator = None
 
-        # Clear all the flags
+        # Clear all the flags and counters
         self._should_stand = False
-        self._should_start_method = False
         self._should_start_ringing_rounds = False
+        self._rounds_left_before_method = None
+        # Set _rounds_left_before_method if we are ringing up-down-in (3 rounds for backstroke
+        # start; 2 for handstroke)
+        if self._do_up_down_in:
+            self._rounds_left_before_method = 2 if self.row_generator.start_stroke().is_hand() else 3
```

Comment:
Yep this looks nicer.  Not trying to be a rust fanboy, but rust replaced the `expr ? expr : expr` syntax with being able to use `if` statements inside expressions, and a statement with no semicolon returns the value.  So this would be:
```rust
self._rounds_left_before_method = if !self._do_up_down_in {
    None
} else if self.row_generator.start_stroke().is_hand() {
    2
} else {
    3
};
```
although tbh you'd probably use a `match` statement in this case.  But anyway enough rambling Wheatley is in Python, and this looks good.